_posts/batlopez/2016-04-30-Nat'sFinalProj.md

```diff
@@ -0,0 +1,31 @@
+
+
+User Interface: Using the starter code provided in class, I was able to create a user interface where the user can first select the way in which they would like their data returned on the screen. They have four options: (1) codes and counts, (2) codes, course descriptions, and counts, (3) codes, course descriptions, counts, and visualizations. I built a function that basically searches for the key word and then call it under each if statement that matches the way the user wants the results reported. The one thing I didn't really have time to figure out was how to add the helper function to the helper code rather then have it as part of the main page. That just makes the main page a bit longer than I'd prefer. 
+
+Some of the major issues I had with my code are the following: 
+
+ISSUE#1: One of the issues I came across that is a small (and annoying bug) is with option one, I made a table to return the codes and counts next to them. Because some of the codes are three letter versus four letter codes, though I was able to write the code with regex to return the whole course code, it is messing up the way the tab works so that the ones with three letter codes doesn't tab as much as the one with four letter codes. This is just aesthetically annoying!  
+
+ISSUE#2: Data file is too large for trinket. I ran the program with another IDE and was able to run the entire code. Right now, its pretty frustrating because I couldn't figure out how to get it to sift through the whole course catalog. I tried creating separate CSV files in tricket and basically breaking it up so the user could select the discipline based on alphabet (so pick courses in departments that start with letters A-D) and then only call and read that file if that one is selected but it slowed trinket down and wouldn't run. I don't really know how to handle this situation. 
```

Comment:
We didn't go over memory efficiency but there are ways to get more control over how your program reads in data so it won't overload like this.  Check them out in your future learnings!